app/templates/client/components/auth(auth)/auth.service.js

```diff
@@ -20,7 +19,7 @@ function AuthService($location, $http, $cookies, $q, appConfig, Util, User) {
      * @param  {Function} callback - optional, function(error, user)
      * @return {Promise}
      */
-    login: function(user, callback) {
+    login: function(user, callback = angular.noop) {
```

Comment:
By design if the users passes in anything then they are opting for async behavior and we return a promise.